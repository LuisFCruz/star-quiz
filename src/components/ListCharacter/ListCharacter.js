import { withStyles } from '@material-ui/core';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

import CardCharacter from '../CardCharacter/CardCharacter';
import Progress from '../Progress/Progress';

const styles = {
  grid: {
    display: 'grid',
    gridGap: '10px',
    gridTemplateColumns: 'repeat(auto-fill, minmax(180px, 1fr))',
    padding: '20px',
    maxWidth: '1000px',
    margin: '0 auto',
  }
}

export class ListCharacter extends Component {
  static propTypes = {
    characters: PropTypes.array,
    maxPage: PropTypes.number,
    finished: PropTypes.bool
  };

  render() {
    const { characters } = this.props;

    if (!characters.length) {
      return <Progress />;
    }
    
    return (
      <div style={styles.grid}>
        {characters.map(character => (
          <CardCharacter key={character.id} character={character} />
        ))}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  const { characters: allCharacter, page, finished } = state;

  const start = (page - 1) * 10;
  const end = (page * 10);
  
  const characters = allCharacter.slice(start, end);
  const maxPage = Math.ceil(allCharacter.length / 10);

  return { characters, maxPage, finished };
}

export default connect(mapStateToProps)(withStyles(styles)(ListCharacter));
