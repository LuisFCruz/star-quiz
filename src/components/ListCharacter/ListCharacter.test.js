import { shallow } from "enzyme";
import React from "react";

import { ListCharacter } from "./ListCharacter";

describe("<ListCharacter />", () => {
  test("smoke test", () => {
    expect(ListCharacter).toBeDefined();
  });

  test("render widthout characters", () => {
    const props = { characters: [] };
    const wrapper = shallow(<ListCharacter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("render width characters", () => {
    const character = { id: "1" };
    const props = { characters: [character] };
    const wrapper = shallow(<ListCharacter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
