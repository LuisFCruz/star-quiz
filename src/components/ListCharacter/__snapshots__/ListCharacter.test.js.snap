// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListCharacter /> render width characters 1`] = `
<div
  style={
    Object {
      "display": "grid",
      "gridGap": "10px",
      "gridTemplateColumns": "repeat(auto-fill, minmax(180px, 1fr))",
      "margin": "0 auto",
      "maxWidth": "1000px",
      "padding": "20px",
    }
  }
>
  <Connect(CardCharacter)
    character={
      Object {
        "id": "1",
      }
    }
    key="1"
  />
</div>
`;

exports[`<ListCharacter /> render widthout characters 1`] = `<Progress />`;
